### PR TITLE
[ENHANCEMENT] add custom field mode support

### PR DESCRIPTION
**Pull request by burtenshaw**
# Description

Closes #<issue_number>

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/tests/factories.py [hunk 1]
```diff
@@ -275,9 +275,7 @@ class ChatFieldFactory(FieldFactory):
 
 
 class CustomFieldFactory(FieldFactory):
-    settings = {
-        "type": FieldType.custom,
-    }
+    settings = {"type": FieldType.custom}
```
Comment (frascuchon) [inline argilla-server/tests/factories.py]: we should add a value for `template` and `advanced`

File: argilla-server/src/argilla_server/api/schemas/v1/fields.py [hunk 2]
```diff
@@ -91,16 +91,19 @@ class ChatFieldSettingsUpdate(BaseModel):
 class CustomFieldSettings(BaseModel):
     type: Literal[FieldType.custom]
     template: str
+    advanced: bool
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/api/schemas/v1/fields.py]: I feel `advanced` a bit wide as naming. Maybe we can add `mode` somewhere: `advanced_mode`?

File: argilla/src/argilla/settings/_field.py [hunk 3]
```diff
@@ -214,6 +214,7 @@ def __init__(
         self,
         name: str,
         template: Optional[str] = "",
+        advanced: Optional[bool] = False,
```
Comment (frascuchon) [inline argilla/src/argilla/settings/_field.py]: Adding some ref in docs would be nice.

**Comment by frascuchon** [comment]
And maybe it makes sense to include here all required changes in the frontend part cc @damianpumar @leiyre

**Comment by burtenshaw** [comment]
> And maybe it makes sense to include here all required changes in the frontend part cc @damianpumar @leiyre

Should we merge these SDK changes on to https://github.com/argilla-io/argilla/pull/5422, and then make the frontend changes there?

**Review by frascuchon — APPROVED** [review]

**Pull request merged**